bee-crypto/src/ternary/sponge/batched_curlp/bct_curlp.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::ternary::{
+    sponge::{
+        batched_curlp::{bct::BCTritBuf, HIGH_BITS},
+        CurlPRounds,
+    },
+    HASH_LENGTH,
+};
+
+pub struct BCTCurlP {
```

Comment:
Does it make sense to implement `Sponge` trait for it?

Reply:
We've been thinking about it but that would mean we need to extend `Sponge` to have something like a `Buffer` and `Slice` type. And change the `BCTBuf/Ref/Mut` to have an actual equivalent of `[usize]` which is a little bit complicated to do given that Rust doesn't allow to have two dynamically sized types on the same struct.